common/core/web/keyboard-processor/src/text/kbdInterface.ts

```diff
@@ -958,5 +958,47 @@ namespace com.keyman.text {
 
       return behavior;
     }
+
+    // Publishes the KeyboardInterface's shorthand API names.
+    // Note that this may need to be called multiple times; a keyboard-processor consumer
+    // may extend or overwrite some of the callbacks after this method's initial call.
+    static __publishShorthandAPI() {
+      // Keyboard callbacks
+      let prototype = this.prototype;
+
+      var exportKBCallback = function(miniName: string, longName: string) {
+        prototype[miniName] = prototype[longName];
+      }
+
+      exportKBCallback('KSF', 'saveFocus');
+      exportKBCallback('KBR', 'beepReset');
+      exportKBCallback('KT', 'insertText');
+      exportKBCallback('KR', 'registerKeyboard');
+      exportKBCallback('KRS', 'registerStub');
+      exportKBCallback('KC', 'context');
+      exportKBCallback('KN', 'nul');
+      exportKBCallback('KCM', 'contextMatch');
+      exportKBCallback('KFCM', 'fullContextMatch');
+      exportKBCallback('KIK', 'isKeypress');
+      exportKBCallback('KKM', 'keyMatch');
+      exportKBCallback('KSM', 'stateMatch');
+      exportKBCallback('KKI', 'keyInformation');
+      exportKBCallback('KDM', 'deadkeyMatch');
+      exportKBCallback('KB', 'beep');
+      exportKBCallback('KA', 'any');
+      exportKBCallback('KDC', 'deleteContext');
+      exportKBCallback('KO', 'output');
+      exportKBCallback('KDO', 'deadkeyOutput');
+      exportKBCallback('KIO', 'indexOutput');
+      exportKBCallback('KIFS', 'ifStore');
+      exportKBCallback('KSETS', 'setStore');
+      exportKBCallback('KLOAD', 'loadStore');
+      exportKBCallback('KSAVE', 'saveStore');
+    }
   }
+
+  (function() {
+    // This will be the only call within the keyboard-processor module.
+    KeyboardInterface.__publishShorthandAPI();
```

Comment:
This call performs the baseline keyboard-API publishing necessary for keyboards to use in headless unit-testing.  Note that at this point in the code, any DOM-based extensions (like those found in [test/KMW's domKbdInterface.ts](https://github.com/keymanapp/keyman/blob/master/web/source/text/domKbdInterface.ts)) do not yet exist!